Compilation fails when resource file is name `Resources.resx`

## Describe the Bug

Compilation fails when resource file is name `Resources.resx`

## Steps to Reproduce

Using ThisAssembly 1.1.3

We have a `Resources.resx` file, set as Embedded resource, and when updating from 1.0.10 to 1.1.3 we get this error at compile:

```
'Resources': member names cannot be the same as their enclosing type
```

This is the file generated by ThisAssembly

```c#
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//
//     ThisAssembly.Resources: 1.1.3
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------
using System;
using System.IO;

partial class ThisAssembly
{
    public static partial class Resources
    {
        /// <summary>
        /// => @"Resources.resx"
        /// </summary>
        public static partial class Resources
        {
            public static byte[] GetBytes() => EmbeddedResource.GetBytes(@"Resources.resx");
            public static Stream GetStream() => EmbeddedResource.GetStream(@"Resources.resx");
        }
    }
}
```

## Expected Behavior

I'd expect the version bump to work as it is a minor version.

## Exception with Stack Trace

```text
1>------ Rebuild All started: Project: REDACTED, Configuration: Debug Any CPU ------
1>C:\Users\redacted\Code\redacted\src\redacted\ThisAssembly.Resource\ThisAssembly.ResourcesGenerator\Resources.g.cs(21,37,21,46): error CS0542: 'Resources': member names cannot be the same as their enclosing type
1>Done building project "REDACTED.csproj" -- FAILED.
```

## Version Info

We are currently on 1.0.10 and trying to upgrade to 1.1.3

## Additional Info

--- a/src/PathSanitizer.cs
+++ b/src/PathSanitizer.cs
@@ -3,10 +3,10 @@
 static class PathSanitizer
 {
     static readonly Regex invalidCharsRegex = new(@"\W");
-    public static string Sanitize(string path)
+    public static string Sanitize(string path, string parent)
     {
         var partStr = invalidCharsRegex.Replace(path, "_");
-        if (char.IsDigit(partStr[0]))
+        if (char.IsDigit(partStr[0]) || partStr == parent)
             partStr = "_" + partStr;
         return partStr;
     }

--- a/src/PathSanitizer.cs
+++ b/src/PathSanitizer.cs
@@ -6,7 +6,9 @@ static class PathSanitizer
     public static string Sanitize(string path, string parent)
     {
         var partStr = invalidCharsRegex.Replace(path, "_");
-        if (char.IsDigit(partStr[0]) || partStr == parent)
+        if (char.IsDigit(partStr[0]))
+            partStr = "_" + partStr;
+        if (partStr == parent)
             partStr = "_" + partStr;
         return partStr;
     }

--- a/src/PathSanitizer.cs
+++ b/src/PathSanitizer.cs
@@ -1,0 +1,13 @@
+﻿using System.Text.RegularExpressions;
+
+static class PathSanitizer
+{
+    static readonly Regex invalidCharsRegex = new(@"\W");
+    public static string Sanitize(string path)
+    {
+        var partStr = invalidCharsRegex.Replace(path, "_");
+        if (char.IsDigit(partStr[0]))
+            partStr = "_" + partStr;
+        return partStr;
+    }
+}

--- a/src/ThisAssembly.Resources/CSharp.sbntxt
+++ b/src/ThisAssembly.Resources/CSharp.sbntxt
@@ -16,7 +16,7 @@
 		/// => @"{{ $0.Path }}"
 		{{~ end ~}}
 		/// </summary>
-		public static partial class {{ $0.Name | string.replace "-" "_" | string.replace " " "_" }}
+		public static partial class {{ $0.Name }}
 		{
 			{{~ if $0.IsText ~}}
 			private static string text;

--- a/src/ThisAssembly.Resources/Model.cs
+++ b/src/ThisAssembly.Resources/Model.cs
@@ -27,14 +27,21 @@ record Area(string Name)
         var parts = basePath.Split(new[] { "\\", "/" }, StringSplitOptions.RemoveEmptyEntries);
         var end = resources.Count == 1 ? ^1 : ^0;
 
+        var parent = "Resources";
         foreach (var part in parts.AsSpan()[..end])
         {
-            var partStr = PathSanitizer.Sanitize(part);
+            var partStr = PathSanitizer.Sanitize(part, parent);
+            parent = partStr;
+
             area.NestedArea = new Area(partStr);
             area = area.NestedArea;
         }
 
-        area.Resources = resources;
+        area.Resources = resources
+            .Select(r => r with
+            {
+                Name = PathSanitizer.Sanitize(r.Name, parent),
+            });
         return root;
     }
 }

--- a/src/ThisAssembly.Resources/Model.cs
+++ b/src/ThisAssembly.Resources/Model.cs
@@ -40,7 +40,4 @@ record Area(string Name)
 }
 
 [DebuggerDisplay("{Name}")]
-record Resource(string Name, string? Comment, bool IsText)
-{
-    public string? Path { get; set; }
-};
+record Resource(string Name, string? Comment, bool IsText, string Path);

--- a/src/ThisAssembly.Resources/Model.cs
+++ b/src/ThisAssembly.Resources/Model.cs
@@ -29,22 +29,13 @@ record Area(string Name)
 
         foreach (var part in parts.AsSpan()[..end])
         {
-            var partStr = SanitizePart(part);
+            var partStr = PathSanitizer.Sanitize(part);
             area.NestedArea = new Area(partStr);
             area = area.NestedArea;
         }
 
         area.Resources = resources;
         return root;
-    }
-
-    static readonly Regex invalidCharsRegex = new(@"\W");
-    static string SanitizePart(string? part)
-    {
-        var partStr = invalidCharsRegex.Replace(part, "_");
-        if (char.IsDigit(partStr[0]))
-            partStr = "_" + partStr;
-        return partStr;
     }
 }
 

--- a/src/ThisAssembly.Resources/ResourcesGenerator.cs
+++ b/src/ThisAssembly.Resources/ResourcesGenerator.cs
@@ -76,10 +76,11 @@ namespace ThisAssembly
                         var name = group.Count() == 1
                             ? Path.GetFileNameWithoutExtension(f.resourceName)
                             : Path.GetExtension(f.resourceName)[1..];
-                        return new Resource(name, f.comment, isText)
-                        {
-                            Path = f.resourceName,
-                        };
+                        return new Resource(
+                            Name: PathSanitizer.Sanitize(name),
+                            Comment: f.comment,
+                            IsText: isText,
+                            Path: f.resourceName);
                     })
                     .ToList();
 

--- a/src/ThisAssembly.Resources/ResourcesGenerator.cs
+++ b/src/ThisAssembly.Resources/ResourcesGenerator.cs
@@ -77,7 +77,7 @@ namespace ThisAssembly
                             ? Path.GetFileNameWithoutExtension(f.resourceName)
                             : Path.GetExtension(f.resourceName)[1..];
                         return new Resource(
-                            Name: PathSanitizer.Sanitize(name),
+                            Name: name,
                             Comment: f.comment,
                             IsText: isText,
                             Path: f.resourceName);

--- a/src/ThisAssembly.Tests/Tests.cs
+++ b/src/ThisAssembly.Tests/Tests.cs
@@ -56,5 +56,9 @@ namespace ThisAssemblyTests
         [Fact]
         public void CanUseSameNameDifferentExtensions()
             => Assert.NotNull(ThisAssembly.Resources.Content.Swagger.swagger_ui.css.GetBytes());
+
+        [Fact]
+        public void CanUseFileInvalidCharacters()
+            => Assert.NotNull(ThisAssembly.Resources.webhook_data.Text);
     }
 }

--- a/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj
+++ b/src/ThisAssembly.Tests/ThisAssembly.Tests.csproj
@@ -12,9 +12,7 @@
   <Import Project="..\*\*.props" />
 
   <ItemGroup>
-    <Compile Remove="..\EmbeddedResource.cs" />
-    <Compile Remove="..\GeneratorExtension.cs" />
-    <Compile Remove="..\IsExternalInit.cs" />
+    <Compile Remove="..\*.cs" />
   </ItemGroup>
 
   <ItemGroup>

